src/foundry/client/data/documents/scene.d.ts

```diff
@@ -122,332 +91,385 @@ declare global {
      */
     getDimensions(): SceneDimensions;
 
+    /** @internal */
+    protected _onClickDocumentLink(event: MouseEvent): any;
+
+    /** @internal */
     protected override _preCreate(
-      data: SceneDataConstructorData,
-      options: DocumentModificationOptions,
-      user: foundry.documents.BaseUser
+      data: BaseScene.ConstructorData,
+      options: { fallback?: boolean; recursive?: boolean },
+      user: BaseUser
     ): Promise<void>;
 
+    /** @internal */
     protected override _onCreate(
-      data: foundry.data.SceneData["_source"],
-      options: DocumentModificationOptions,
+      data: BaseScene.Source,
+      options: { fallback?: boolean; recursive?: boolean },
       userId: string
     ): void;
 
+    /** @internal */
     protected override _preUpdate(
-      changed: DeepPartial<SceneDataConstructorData>,
-      options: DocumentModificationOptions,
-      user: foundry.documents.BaseUser
+      data: BaseScene.UpdateData,
+      options: { fallback?: boolean; recursive?: boolean },
+      user: BaseUser
     ): Promise<void>;
 
+    /** @override */
     protected override _onUpdate(
-      changed: DeepPartial<foundry.data.SceneData["_source"]> & Record<string, unknown>,
-      options: DocumentModificationOptions,
+      data: DeepPartial<fields.SchemaField.PersistedType<BaseScene.SchemaField["fields"], {}>>,
```

Comment:
This could be `DeepPartial<BaseScene.Source>`.

Reply:
🤩

Reply:
Updated